### DISCUSSION TITLE
Failed to JIT compile expression

### DISCUSSION THREAD

**Discussion by heinono1** [idaholab/moose#24298] [Q&A General] [answered]
Hello,

I got a message while executing that I have never seen before:
```
"*** Info ***
Failed to JIT compile expression, falling back to byte code interpretation." This seems to lead to a fatal error later:
"JIT compile failed.
In file included from /opt/ohpc/pub/compiler/gcc/9.4.0/include/c++/9.4.0/x86_64-pc-linux-gnu/bits/c++config.h:528,
                 from /opt/ohpc/pub/compiler/gcc/9.4.0/include/c++/9.4.0/cmath:41,
                 from ./tmp_jit_mpl4ep.cc:2:
/opt/ohpc/pub/compiler/gcc/9.4.0/include/c++/9.4.0/x86_64-pc-linux-gnu/bits/os_defines.h:39:10: fatal error: features.h: No such fileor directory
   39 | #include <features.h>"
```
The weird thing is that I run the same code on another (Linux) machine using the same gnu compiler suite (I think) and there I do not get this message and error. 

This issue seems to be not the same as JIT compile error when "too many files open" using ParsedMaterials (tophmatthews #9549). I am using ParsedMaterials,  but only three of them that are also linked: the third one uses the second one which uses the first one.

Any ideas of what is going on here?

**Comment by GiudGiud** [comment]
Can you share the parsed expression here?

And you have write access to the folder this is running in / permissions set correctly right

**Reply by heinono1** [reply]
Hi Guillaume,

Yes, I have write access to the folder.

Below are the ParsedMaterials. The quantities mur, aparam, Ms are GenericConstantMaterials defined in preceding blocks:
````
   [./hnorm]
    type = ParsedMaterial
    property_name = h_norm
    coupled_variables = 'Hext_x Hext_y Hext_z'#
    expression = 'sqrt((Hext_x)^2+(Hext_y)^2+(Hext_z)^2)'
    execute_on = 'final'
  [../]
  [./Ha]
    type = ParsedMaterial
    property_name = Ha
    material_property_names = 'mur Ms h_norm'
    expression = 'max(0.,(mur -1.)*h_norm/Ms)'
    execute_on = 'final'
  [../]
    [./permeability]
    type = ParsedMaterial
    property_name = chi
    material_property_names = 'Ms aparam Ha h_norm'
    expression = '0.5*Ms*(Ha +1.-sqrt((Ha+1.)^2-4.*Ha*(1.-aparam)))/((1.-aparam))'
    execute_on = 'final'
  [../]
```

**Reply by heinono1** [reply]
Note that copy and paste missed some * signs for multiplication in the parsed expressions.

**Reply by GiudGiud** [reply]
formatting got them back

**Reply by GiudGiud** [reply, marked as answer]
for the last one, can you try:
`     material_property_names = 'Ms aparam Ha'`
you dont use h_norm

for all, can you please try removing this `execute_on = final`. I ve never done that on Material

**Reply by heinono1** [reply]
Hi Guillaume. Yeah it looks like the "execute_on" is causing trouble. Removing that line for the ParsedMaterials seems to have made the problem go away. Interesting that the two different computer platforms gave different errors. On my linux workstation (Ubuntu 20.04 and gnu compilers) I did not get a warning about JIT compile and subsequent crash - instead the execution crashed because "unused parameter 'Materials/permeability/execute_on' "
